int32 x_distance
float32 y_distance
int32 z_distance